app/src/lib/stores/repositories-store.ts

```diff
@@ -399,6 +405,45 @@ export class RepositoriesStore extends BaseStore {
           gitHubRepositoryID: updatedGitHubRepo.dbID,
         })
 
+        if (enableBranchProtectionWarning()) {
+          const repoId = updatedGitHubRepo.dbID!
+
+          // This update flow is organized into two stages:
+          //
+          // - update the in-memory cache
+          // - update the underyling database state
+          //
+          // This should ensure any stale values are not being used, and avoids
+          // the need to query the database while the results are in memory.
+
+          const prefix = getKeyPrefix(repoId)
+
+          for (const key of this.branchProtectionCache.keys()) {
+            // invalidate any cached entries belonging to this repository
+            if (key.startsWith(prefix)) {
```

Comment:
Ordinarily I'd be hesitant with a linear search here but given that I expect that the number of protected branches will be tiny (i.e. very rarely more than 1 per repository and most of the time none at all) I think this is fine. Even if we 10x our expectations and 10x the average number of repositories this number should still be manageable.